Sources/Drain.swift

```diff
@@ -33,9 +34,6 @@ public final class Drain: DataRepresentable, Stream {
 
     public init(_ buffer: Data) {
         self.buffer = buffer
-        if buffer.bytes.isEmpty {
-            close()
-        }
```

Comment:
@paulofaria re: `init(closed: Bool)`. I'm remembering now that I just removed this buffer empty check to auto-close and it worked fine. I don't know if this is important to keep around. But, if it's not, this fixes the problem without need another init.